src/plans/D3MCompoundPlan.sol

```diff
@@ -0,0 +1,187 @@
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+////////////////////////////////////////////////////////////////////////////////
+//                                                                            //
+//    Methodology for Calculating Target Supply from Target Rate Per Block    //
+//                                                                            //
+////////////////////////////////////////////////////////////////////////////////
+
+// apy to rate per block off-chain conversion info:
+//         apy = ((1 + (borrowRatePerBlock / WAD) * blocksPerDay) ^ 365 - 1) * 100
+// (0)     borrowRatePerBlock = ((((apy / 100) + 1) ^ (1 / 365) - 1) / blocksPerDay) * WAD
+//         ATTOW the formula matches the borrow apy on (https://compound.finance/markets/DAI) using blocksPerDay = 6570
+//
+// https://github.com/compound-finance/compound-protocol/blob/master/contracts/BaseJumpRateModelV2.sol#L95
+//
+// util > kink:
+//         normalRate = kink * multiplierPerBlock / WAD + baseRatePerBlock;
+//         targetInterestRate = normalRate + jumpMultiplierPerBlock * (util - kink) / WAD
+//         util * jumpMultiplierPerBlock = (targetInterestRate - normalRate) * WAD + kink * jumpMultiplierPerBlock
+// (1)     util = kink + (targetInterestRate - normalRate) * WAD / jumpMultiplierPerBlock
+//
+// util <= kink:
+//         targetInterestRate = util * multiplierPerBlock / WAD + baseRatePerBlock
+// (2)     util = (targetInterestRate - baseRatePerBlock) * WAD / multiplierPerBlock
+//
+// https://github.com/compound-finance/compound-protocol/blob/master/contracts/BaseJumpRateModelV2.sol#L79
+//
+//         util = borrows * WAD / (cash + borrows - reserves);
+// (3)     cash + borrows - reserves = borrows * WAD / util
+
+pragma solidity ^0.8.14;
+
+import "./ID3MPlan.sol";
+
+// https://github.com/compound-finance/compound-protocol/blob/3affca87636eecd901eb43f81a4813186393905d/contracts/CErc20.sol#L14
+interface CErc20Like {
+    function totalBorrows()           external view returns (uint256);
+    function totalReserves()          external view returns (uint256);
+    function interestRateModel()      external view returns (address);
+    function getCash()                external view returns (uint256);
+    function underlying()             external view returns (address);
+}
+
+// https://github.com/compound-finance/compound-protocol/blob/3affca87636eecd901eb43f81a4813186393905d/contracts/BaseJumpRateModelV2.sol#L10
+interface InterestRateModelLike {
+    function baseRatePerBlock()       external view returns (uint256);
+    function kink()                   external view returns (uint256);
+    function multiplierPerBlock()     external view returns (uint256);
+    function jumpMultiplierPerBlock() external view returns (uint256);
+}
+
+contract D3MCompoundPlan is ID3MPlan {
```

Comment:
Has there been any research done on whether there's any risk of reentrancy on the CToken being used to manipulate the results? Unfotunately the reentrancy guard variable on a CToken is not public...and I think the only functions marked non-reentrant are not state-modifying (meaning we couldn't have `view` functions here if we called one of them). There may be no issue anyway, but was curious if anyone besides me had thought about it.

Reply:
I'm not aware of such research (probably not for AAVE as well). We should look into that.
Can you think of what an attacker has to gain other than causing unintended (and possibly multiple) winds/unwinds?

Reply:
I need to do deeper research into know what the worst-case consequences are. One thing to consider is that other contracts in the future might call the plan's `view` functions and use the results in various ways--it's hard to predict what the consequences could be for such contracts of manipulation.